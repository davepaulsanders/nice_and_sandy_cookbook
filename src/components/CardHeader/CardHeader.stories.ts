import type { Meta, StoryObj } from '@storybook/react';
 
import CardHeader from './CardHeader';
 
const meta: Meta<typeof CardHeader> = {
  component: CardHeader,
};
 
export default meta;
type Story = StoryObj<typeof CardHeader>;
export const Card: Story = {args: {src: "https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg", alt: "test"}};
