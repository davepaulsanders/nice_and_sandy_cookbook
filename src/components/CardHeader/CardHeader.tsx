import { Recipe } from "../../types/types"

interface CardHeaderProps extends Recipe {}
	
const CardHeader = ({ img, alt, ..._rest }: CardHeaderProps) => {
	const defaultImgUrl = "https://images.unsplash.com/photo-1645802734096-07f97fcd8bf2?q=80&w=1740&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
	return (
		<div className="mt-4 w-11/12">
		<img className="h-[27rem] w-full object-cover" 
		src={img || defaultImgUrl} alt={alt} />
		</div>
		)
}
export default CardHeader
