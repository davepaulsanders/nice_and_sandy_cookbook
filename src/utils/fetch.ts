export const fetchData = async (url: string) => {
	const resp = await fetch(url)
	if (resp.status !== 200) throw new Error(`Request failed with ${resp.status}`)
	const data = resp.json()
	return data
}
