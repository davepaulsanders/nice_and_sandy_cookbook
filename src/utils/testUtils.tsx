import RecipeCard from '../components/RecipeCard/RecipeCard';
import { ReactElement } from "react"
import { vi } from "vitest"
import { Recipe } from "../types/types"

const generateRecipes = () => {
const recipes: Recipe[] = []
for (let i = 0; i < 10; i++) {
	recipes.push({
	id:i + 1,
	img:"https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg",
	href:"https://www.feastingathome.com/vegan-fajitas/",
	label:`Vegan Fajitas ${i + 1}`,
	alt:"test",
	category:"ENTREE",
	category_id:1,
	is_pinned: false,
	})
}
return recipes
}
const generateRecipeCards = () => {
const recipeCards: ReactElement[] = []
for (let i = 0; i < 10; i++) {
	recipeCards.push(
	<RecipeCard
	id={i + 1}
	img="https://www.feastingathome.com/wp-content/uploads/2024/10/Vegan-Fajitas-13.jpg"
	href="https://www.feastingathome.com/vegan-fajitas/"
	label="Vegan Fajitas 2"
	alt="test"
	category="ENTREE"
	category_id={1}
	is_pinned= {false}
	recipes={generateRecipes()}
	setRecipes={vi.fn()}
	/>)
	}
	return recipeCards
}
export { generateRecipeCards, generateRecipes }
