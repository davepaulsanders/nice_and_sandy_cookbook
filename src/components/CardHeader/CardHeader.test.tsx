import CardHeader from "./CardHeader" 
import { expect, test } from 'vitest'
import { render } from '@testing-library/react'
import '@testing-library/dom'
import { generateRecipes } from "../../utils/testUtils"
const recipes = generateRecipes()
test('<CardHeader /> renders', () => {
	const { container } = render(<CardHeader  {...recipes[0]}/>)
	expect(container).toMatchSnapshot()
})
