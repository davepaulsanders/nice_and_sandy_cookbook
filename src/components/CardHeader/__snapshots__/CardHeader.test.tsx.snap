// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CardHeader /> renders 1`] = `
<div>
  <div
    class="w-11/12"
  >
    <img
      alt="test"
      class="h-[27rem] w-full object-cover"
      src="https://fastly.picsum.photos/id/454/200/300.jpg?hmac=wGXBDB3HURz2isRdLrgggeWdD1yO3rdX4B3-jlzRncg"
    />
  </div>
</div>
`;
